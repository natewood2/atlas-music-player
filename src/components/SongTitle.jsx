import useSongs from '../hooks/useSongs';
import { useState } from 'react';
import React from 'react';

export default function SongTitle({ song }) {
  if (!song) return null;

  return (
    <div>
      <p className="text-lg dark:text-zinc-200 font-bold">{song.title}</p>
      <p className="text-[#64748B] dark:text-gray-500">{song.artist}</p>
    </div>
  );
}
