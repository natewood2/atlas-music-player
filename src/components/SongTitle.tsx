import React from 'react';

interface SongTitleProps {
  song?: {
    title: string;
    artist: string;
  };
}

export default function SongTitle({ song }: SongTitleProps) {
  if (!song) {
    return (
      <div className="mt-8 text-center">
        {/* I dont think this is working */}
        <div className="h-6 w-48 bg-gray-200 animate-pulse rounded-md mb-2"></div>
        <div className="h-4 w-32 bg-gray-200 animate-pulse rounded-md"></div>
      </div>
    );
  }

  return (
    <div className="mt-8 text-center">
      <h2 className="text-2xl font-bold mb-2">{song.title}</h2>
      <p className="text-gray-500">{song.artist}</p>
    </div>
  );
}