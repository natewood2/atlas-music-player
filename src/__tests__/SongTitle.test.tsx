import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import SongTitle from '../components/SongTitle';

it('renders SongTitle', () => {
  const song = { title: "Test Song", artist: "Test Artist" };
  const { container } = render(<SongTitle song={song} />);
  expect(container).toMatchSnapshot();
});