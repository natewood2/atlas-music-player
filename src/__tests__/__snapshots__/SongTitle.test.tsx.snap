// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders SongTitle 1`] = `
<div>
  <div
    class="mt-8 text-center"
  >
    <h2
      class="text-2xl font-bold mb-2"
    >
      Test Song
    </h2>
    <p
      class="text-gray-500"
    >
      Test Artist
    </p>
  </div>
</div>
`;
